fast64_internal/oot/oot_model_classes.py

```diff
@@ -1,191 +1,531 @@
+                name,
+                texProp.tex_format,
+                texProp.ci_format,
+                True,
+                sharedPalette,
+            )
+            existingFPalette = self.validateCIFlipbook(existingFPalette, alreadyExists, fPalette, flipbookTexture)
+            fImages.append(fImage)
+
+            # do this here to check for modified names due to repeats
+            flipbook.textureNames.append(fImage.name)
+
+        self.addFlipbookWithRepeatCheck(flipbook)
+
+        # print(f"Palette length for {sharedPalette.name}: {len(sharedPalette.palette)}")
+        firstImage = flipbookProp.textures[0].image
+        self.processedFlipbooks[firstImage] = [flipbookTex.image for flipbookTex in flipbookProp.textures]
+
+        if existingFPalette == False:
+
+            palFormat = texProp.ci_format
+            fPalette = saveOrGetPaletteOnlyDefinition(
+                fMaterial,
+                self,
+                firstImage,
+                sharedPalette.name,
+                texProp.tex_format,
+                palFormat,
+                True,
+                sharedPalette.palette,
+            )
+
+            # using the first image for the key, apply paletteKey to all images
+            # while this is not ideal, its better to us an image for the key as
+            # names are modified when duplicates are found
+            paletteKey = (firstImage, (palFormat, "PAL"))
+            for fImage in fImages:
+                fImage.paletteKey = paletteKey
+        else:
+            fPalette = existingFPalette
+
+        self.modifyDLForCIFlipbook(fMaterial, fPalette, texProp)
+
+    def processFlipbookNonCI(self, fMaterial: FMaterial, flipbookProp: Any, texProp: TextureProperty):
+        flipbook = TextureFlipbook(flipbookProp.name, flipbookProp.exportMode, [])
+        for flipbookTexture in flipbookProp.textures:
+            if flipbookTexture.image is None:
+                raise PluginError(f"Flipbook for {fMaterial.name} has a texture array item that has not been set.")
+            # print(f"Texture: {str(flipbookTexture.image)}")
+            name = (
+                flipbookTexture.name
+                if flipbookProp.exportMode == "Individual"
+                else self.name + "_" + flipbookTexture.image.name + "_" + texProp.tex_format.lower()
+            )
+            fImage = saveOrGetTextureDefinition(
+                fMaterial,
+                self,
+                flipbookTexture.image,
+                name,
+                texProp.tex_format,
+                True,
+            )
+
+            # do this here to check for modified names due to repeats
+            flipbook.textureNames.append(fImage.name)
+        self.addFlipbookWithRepeatCheck(flipbook)
+
+    def onMaterialCommandsBuilt(self, fMaterial, material, drawLayer):
+        # handle dynamic material calls
+        gfxList = fMaterial.material
+        matDrawLayer = getattr(material.flipbookGroup, drawLayer.lower())
```

Comment:
I think this line was just a typo, I changed this line back to what it was originally and it seems to have fixed the error I was getting.
```suggestion
        matDrawLayer = getattr(material.ootMaterial, drawLayer.lower())
```

Reply:
Whoops, you're right. Somehow slipped through the cracks.